src/routes/[id]/+page.svelte

```diff
@@ -1,29 +1,36 @@
 <script>
   import { onMount } from "svelte";
-  import { page } from "$app/stores"; // Import $page from $app/stores
+  import { page } from "$app/stores";
   import { HighlightAuto } from "svelte-highlight";
   import atomOneDark from "svelte-highlight/styles/atom-one-dark";
   import ClipboardJS from "clipboard";
   import toast, { Toaster } from "svelte-french-toast";
   import { jsPDF } from "jspdf";
+
   let id = null;
   let paste = null;
   let code = null;
+  let errorMessage = "";
   const doc = new jsPDF();
 
   onMount(async () => {
-    id = $page.params.id; // Now $page is correctly imported and can be used
-    console.log(`fetching /api?id=${id}`);
-    fetch(`/api?id=${id}`)
-      .then((res) => res.json())
-      .then((data) => {
-        // console.log("Fetched paste:", data.id);
-        paste = data.id;
-        new ClipboardJS(".btn-clip");
-      })
-      .catch((error) => {
-        console.log("Error fetching paste:", error);
-      });
+    id = $page.params.id;
+    console.log(`fetching /api/paste/${id}`);
+
+    try {
+      const res = await fetch(`/api/paste/${id}`);
+      if (!res.ok) {
+        throw new Error("Invalid or non-existent paste ID.");
+      }
+      const data = await res.json();
+      paste = data;
+      code = paste.text;
```

Comment:
The old method was okay here 
fetch(`/api?id=${id}`)
      .then((res) => res.json())
      .then((data) => {
        // console.log("Fetched paste:", data.id);
        paste = data.id;
        new ClipboardJS(".btn-clip");
      })